server/pulp/plugins/util/publish_step.py

```diff
@@ -824,6 +824,34 @@ def process_main(self):
         shutil.move(os.path.join(self.source_dir, tar_file_name), self.publish_file)
 
 
+class CreatePulpManifestStep(Step):
+    """
+    This will create a PULP_MANIFEST file in the specified directory. This step should be used when
+    the checksums of the files are not already known, because it will read and calculate new
+    checksums for each one.
+
+    If you already know the SHA256 checksums of the files going in the manifest, see an example
+    in the FileDistributor that creates this file in a different way.
+    """
+    def __init__(self, target_dir):
```

Comment:
What if this took an optional argument of a generator of (file_path, sha256_checksum, size) tuples that would be used instead of the make_manifest_for_dir method if provided? That way all places that need a  manifest could use the same code path to generate it.